colabs/wandb-artifacts/WandB_Artifacts_Time_to_live_TTL_Walkthrough.ipynb

```diff
@@ -0,0 +1,382 @@
+   "metadata": {},
+   "source": [
+    "# Updating/Retroactively Setting TTL on Artifacts\n",
+    "In our hypothetical organization we've been given approval to retain a specific version of our data indefinitely. We've also been given approval to extend the retention date of an additional dataset. Below we'll:\n",
+    "\n",
+    "- Extend the TTL of an Artifact tagged with the `extended` alias\n",
+    "- Remove the TTL of an Artifact tagged with the `compliant` alias\n",
+    "- Programmatically check the status of these two Artifacts"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with wandb.init(entity=\"wandb-smle\", project=\"artifacts-ttl-demo\", job_type=\"modify-ttl\") as run:\n",
+    "  extended_art = run.use_artifact(\"wandb-smle/artifacts-ttl-demo/real-raw:extended\")\n",
+    "  extended_art.ttl = timedelta(days=365)  # Delete in a year\n",
+    "  extended_art.save()\n",
+    "\n",
+    "  compliant_art = run.use_artifact(\"wandb-smle/artifacts-ttl-demo/real-raw:compliant\")\n",
+    "  compliant_art.ttl = None\n",
+    "  compliant_art.save()\n",
+    "\n",
+    "  print(extended_art.ttl)\n",
+    "  print(compliant_art.ttl)\n",
+    "\n",
+    "  run.finish()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Use W&B Import/Export API to Iterate Artifact Versions and Set TTL\n",
+    "Let's say we've received approval to retain all of the data within a given Artifact and we'd like to remove all TTL policies for every version of an Artifact. Below we:\n",
+    "\n",
+    "- Use the W&B API to get a list of all Runs in a project\n",
+    "- Get a list of all versions of a specific Artifact (e.g. `fake-raw`)\n",
+    "- Iterate over each  version and remove any existing TTL policy associated with the version"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Artifact metadata extraction\n",
+    "api = wandb.Api()\n",
+    "\n",
+    "# Define entity and project\n",
+    "entity, project = \"wandb-smle\", \"artifacts-ttl-demo\"\n",
+    "\n",
+    "runs = api.runs(entity + \"/\" + project)\n",
+    "\n",
+    "version_names = []\n",
+    "for run in runs:\n",
+    "    for artifact in iter(run.logged_artifacts()):\n",
+    "        if \"fake-raw\" in artifact.name:\n",
+    "          # Can be edited to just display individual elements\n",
+    "          version_names.append(f\"{artifact.name}/{artifact.version}\")\n",
+    "\n",
+    "with wandb.init(entity=\"wandb-smle\", project=\"artifacts-ttl-demo\", job_type=\"modify-ttl\") as run:\n",
+    "  for version in version_names:\n",
+    "    version_art = run.use_artifact(f\"wandb-smle/artifacts-ttl-demo/{'/'.join(version.split('/')[:-1])}\")\n",
+    "    version_art.ttl = None\n",
+    "    version_art.save()\n",
+    "    print(version_art.ttl)\n"
```

Comment:
At the end of this section which shows users how to add TTLs to all artifacts in a project, can we add a pointer to docs where we inform users how admins can set a TTL default for the team if their use case requires it (since this section refers to mass-setting of TTL, the team default feature we be a good thing to highlight in case they want to set a default once rather than programmatically iterating). Suggestion:

_To apply a TTL policy to all artifacts within a team's projects, team admins can set default TTL policies for their team. The default will be applied to both existing and future artifacts logged to projects as long as no custom policies have been set. To learn more about configuring a team default TTL, visit [this](https://docs.wandb.ai/guides/artifacts/ttl#set-default-ttl-policies-for-a-team) section of the W&B documentation._